src/RecurlyPreprocess.php

```diff
@@ -0,0 +1,232 @@
+        'setup_fee' => $setup_fee_amount,
+        'amount' => $unit_amount,
+        'plan_interval' => recurly_format_price_interval($unit_amount, $plan->plan_interval_length, $plan->plan_interval_unit, TRUE),
+        'trial_interval' => $plan->trial_interval_length ? recurly_format_price_interval(NULL, $plan->trial_interval_length, $plan->trial_interval_unit, TRUE) : NULL,
+        'signup_url' => recurly_url('subscribe', [
+          'entity_type' => $entity_type,
+          'entity' => $entity,
+          'plan_code' => $plan_code,
+          'currency' => $currency,
+        ]),
+        'change_url' => $current_subscription ? recurly_url('change_plan', [
+          'entity_type' => $entity_type,
+          'entity' => $entity,
+          'subscription' => $current_subscription,
+          'plan_code' => $plan_code,
+        ]) : NULL,
+        'selected' => FALSE,
+      ];
+
+      // If we have a pending subscription, make that its shown as selected
+      // rather than the current active subscription. This should allow users to
+      // switch back to a previous plan after making a pending switch to another
+      // one.
+      foreach ($subscriptions as $subscription) {
+        if (!empty($subscription->pending_subscription)) {
+          if ($subscription->pending_subscription->plan->plan_code === $plan_code) {
+            $variables['filtered_plans'][$plan_code]['selected'] = TRUE;
+          }
+        }
+        elseif ($subscription->plan->plan_code === $plan_code) {
+          $variables['filtered_plans'][$plan_code]['selected'] = TRUE;
+        }
+      }
+    }
+
+    // Check if this is an account that is creating a new subscription.
+    $variables['expired_subscriptions'] = FALSE;
+    $account = recurly_account_load(['entity_type' => $entity_type, 'entity_id' => $entity->id()]);
+    if ($account) {
+      $variables['expired_subscriptions'] = empty($subscriptions);
+    }
+  }
+
+  /**
+   * Implements hook_preprocess_recurly_subscription_cancel_confirm().
+   *
+   * @param array $variables
+   *   Theme variables to preprocess.
+   */
+  public function preprocessRecurlySubscriptionCancelConfirm(array &$variables) {
+    $variables['subscription'] = $variables['form']['#subscription'];
+    parse_str($this->getRequest()->getQueryString(), $query_array);
+    $variables['past_due'] = isset($query_array['past_due']) && $query_array['past_due'] === '1';
+  }
+
+  /**
+   * Implements hook_preprocess_recurly_invoice_list().
+   *
+   * @param array $variables
+   *   Theme variables to preprocess.
+   */
+  public function preprocessRecurlyInvoiceList(array &$variables) {
+    $invoices = $variables['invoices'];
+    $entity_type = $variables['entity_type'];
+    $entity = $variables['entity'];
+
+    $header = [t('Number'), t('Date'), t('Total')];
+    $rows = [];
+    foreach ($invoices as $invoice) {
+      $status = '';
+      if ($invoice->state === 'past_due') {
```

Comment:
@markdorison Might be worth putting in a todo to switch to state constants, which will be available once the Format Manager service is merged.

```php
  if ($invoice->state === RECURLY_STATE_PAST_DUE) {
```